symengine/polynomial.cpp

```diff
@@ -566,25 +556,20 @@ RCP<const UnivariatePolynomial> add_uni_poly(const UnivariatePolynomial &a,
     } else {
         var = a.get_var();
     }
-    for (const auto &it : a.get_dict())
-        dict[it.first] = it.second;
-    for (const auto &it : b.get_dict())
-        dict[it.first] += it.second;
+    UnivariateExprPolynomial dict = a.get_dict2();
+    dict += b.get_dict2();
```

Comment:
Similarly, here the naming is confusing. You can change the name of this function too.